Ical.Net/Utility/CollectionHelpers.cs

```diff
@@ -66,18 +62,21 @@ public static bool Equals<T>(IEnumerable<T> left, IEnumerable<T> right, bool ord
             return left.SequenceEqual(right);
         }
 
+        // No multiple enumerations
+        var rightArray = right as T[] ?? right.ToArray();
```

Comment:
Eager iteration would only help if right/left would be generators, which doesn't seem to be the case in any of today's uses, otherwise it would rather add overhead.